### ISSUE TITLE
Setting language for text output of buttons

### ISSUE THREAD

**Issue by sonicpunk** [Jako/AjaxUpload#19]
Where can you set what language the snippet AjaxUpload should use if you are NOT using the cultureKey and contexts for different languages?

In my situation I have a lang variable that holds either a 'en' or a 'de' as a value.

AjaxUpload Version 1.5.3
MODX 2.2.16

**Comment by Jako** [comment]
The snippet has a &language property.

**Comment by sonicpunk** [comment]
Yes, I tried that but it still shows German text, even though the language variable gets an 'en' value.

**Comment by Jako** [comment]
Will check that.

**Comment by sonicpunk** [comment]
@Jako Any Luck? Thanks!

**Comment by sonicpunk** [comment]
@Jako sorry to pester you... any change here? My client is waiting :) Sorry for the pressure!

**Comment by Jako** [comment]
I have sent you a message on slack.

**Comment by sonicpunk** [comment]
@Jako This is my snippet call:

[[!AjaxUpload?
    &uid=`files`
    &allowedExtensions=`pdf,ai,eps,jpg,JPG,png,PDF,PNG,EPS`
    &maxFilesizeMb=`5`
    &maxFiles=`5`
    &language=`[[*lang]]`
    ]]

**Comment by sonicpunk** [comment]
@Jako The problem with how this script uses the lexicon is connected with this github issue:
https://github.com/modxcms/revolution/issues/11240

The initial language setting here on line 140 always yielded an empty value. I replaced it with a general getOption call to get the language and was able to get the language saved to the language variable. 
After this, I was able to write the language value to the MODX log, but the lexicon->load call was still not loading the lexicon. That is where the above issue comes into play. I needed to set the cultureKey setting like this to get it to work (before we call the modx->lexicon->load):

        $this->modx->setOption('cultureKey', $language);
        $this->modx->lexicon->load($language.':ajaxupload:default');
This solved the issue.
